code-challenges.test.js

```diff
@@ -15,42 +18,222 @@
+// var more = Array.from(some)
+// console.log(some)
+// console.log(more)
+
+
+// **** I have tried som many things. Spent more than 4+ hours on this one. I can get eahc awnser, but it will only output one array and leave the other one empty. So righ tnow it only outputs fibLength1 but not the fibLength2. But I went back to comment out fibLength1 and fibLength2 shows. I can't figure out why it won't show me both at the samw time
+
 
 // --------------------2) Create a function that takes in an array and returns a new array of only odd numbers sorted from least to greatest.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
+// crate a describe it test with the name of odds and how the expects out come of each arra with only odds from least to greatest.
+
+    // describe("odds", () => {
+    //     // describes what the outcome is goingot be 
+    //     it("expects out come of each arra with only odds from least to greates", () => {
+    //         // write the arrays 
+    //         const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
+    //         const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+    //         // write the expected outcome of the arrays with only odds from least to greatest.
+    //         expect(odds(fullArr1)).toEqual([-9, 7, 9, 199])
+    //         expect(odds(fullArr2)).toEqual([-823, 7, 23])
+    //     })
+    // })
+
+
+
+
+
+// const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
 // Expected output: [-9, 7, 9, 199]
 
-const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+// const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
 // Expected output: [-823, 7, 23]
 
+// I failed the test.
+// ● odds › expects out come of each arra with only odds from least to greates
+
+// ReferenceError: odds is not defined
+
 
 // b) Create the function that makes the test pass.
 
+    // create a function called odds
+    // the function will take in an empty array
+    // we will create a new variabel with ans use  the .filter() from the hof to to call all the odd numbers from condicional to call only the odds numbers whch will be the ones divisible by 3
+    // once we have the new numbers, use a sort to to order the numbers from least to greatest
+    // output should be [-9, 7, 9, 199] and [-823, 7, 23]
+
+
+    // const odds = (array) => {
+    //     let oddArr = array.filter(value => {
+    //         if(typeof(value) % 3 === 0){
+    //                 return array.sort((a, b) => a - b)
+    //         }
+    //     })
+    //         return oddArr
+    //   }
+
+
+    //   I've looked up why I am gettin an extra array on this one too and I can't find anythng. Maybe my jest is broken or may have a bug. If you guys see that I am actually missing soemthing, please let me know and I will fix it.
```

Comment:
Review what typeof() returns. It looks like you want a number in your conditional statement but the typeof() does not do that. 
https://www.w3schools.com/js/js_typeof.asp